spring-cloud-kubernetes-test-support/src/main/java/org/springframework/cloud/kubernetes/integration/tests/commons/K8SUtils.java

```diff
@@ -195,6 +211,41 @@ public void waitForDeployment(String deploymentName, String namespace) {
 				.until(() -> isDeploymentReady(deploymentName, namespace));
 	}
 
+	public void waitForIngress(String ingressName, String namespace) {
```

Comment:
instead of retrying hitting an endpoint, before ingress is up and running, imho: the idiomatic way to do it is to wait for ingress to be ready and only then hit an endpoint. 

This method does exactly that : waits for ingress